test/cuda.jl

```diff
@@ -24,6 +24,17 @@ end
   
 end
 
+@testset "un-broadcasting *, / with mapreduce" begin
+  cu12 = cu(Float32[1,2])
+  @test gradient((x,y) -> sum(x .* y), cu12, 5) == ([5, 5]), 3)
```

Comment:
```suggestion
  @test gradient((x,y) -> sum(x .* y), cu12, 5) == ([5, 5], 3)
```

Reply:
Oops, thanks. 

I've also now removed the tests like `gradient((x,y) -> sum(z -> z, x .* y), cu12, 5) == ([5, 5], 3)`, as these seem to give errors before this PR:
```
julia> gradient((x,y) -> sum(identity, x .* y), cu12, 5)  # same with z->z
ERROR: Compiling Tuple{typeof(task_local_storage), GPUArrays.var"#24#27"{CuArray{Float32, 1}}, Symbol, GPUArrays.ScalarIndexing}: try/catch is not supported.
```